### PR TITLE
remove all unused variables and imports

### PR DESCRIPTION
**Pull request by garrensmith**
Enable the eslint `no-unused-vars`. This just makes our code a little neater.

### CODE REVIEW

File: app/core/routeObject.js [hunk 1]
```diff
@@ -10,8 +10,7 @@
 // License for the specific language governing permissions and limitations under
 // the License.
 
-import FauxtonAPI from "./base";
-import React from "react";
+//import React from "react";
```
Comment (millayr) [inline app/core/routeObject.js]: I'd suggest deleting this versus leaving it commented.

File: app/addons/replication/actions.js [hunk 2]
```diff
@@ -231,15 +231,15 @@ const hideConflictModal = () => {
...
-};
+// const updateUsernameAndPassword = (username, password) => {
+//   FauxtonAPI.dispatch({
+//     type: ActionTypes.REPLICATION_USERNAME_PASSWORD,
+//     options: {
+//       username,
+//       password
+//     }
+//   });
+// };
```
Comment (millayr) [inline app/addons/replication/actions.js]: I'd recommend deleting this versus leaving it commented.  Seems like other unused functions were deleted elsewhere in the PR.

File: Gruntfile.js [hunk 3]
```diff
@@ -77,19 +75,19 @@ module.exports = function (grunt) {
...
+  //   // against a remote CouchDB-compatible server.
+  //   var defaults = {
+  //     port: helper.devServerPort,
+  //     proxy: {
+  //       target: helper.couch
+  //     }
+  //   };
+
+  //   return initHelper.readSettingsFile().couchserver || defaults;
+  // }();
```
Comment (millayr) [inline Gruntfile.js]: I'd recommend deleting this versus commenting

**Comment by millayr** [comment]
This is exciting stuff!  So many unused imports and variables removed!  Just a few comments.

**Comment by millayr** [comment]
+1

**Comment by garrensmith** [comment]
merged

**Pull request closed without merging**